src/common/GeometryCommon/GeometryDecorator.tsx

```diff
@@ -162,10 +156,59 @@ export class GeometryDecorator implements Decorator {
         builder.addLineString(geometry.points);
       } else if (geometry instanceof Loop) {
         builder.addLoop(geometry);
+        if (styledGeometry.edges) {
+          // Since decorators don't natively support visual edges,
+          // We draw them manually as lines along each loop edge/arc
+          builder.setSymbology(ColorDef.black, ColorDef.black, 3);
+          const curves = geometry.children;
+          curves.forEach((value) => {
+            if (value instanceof LineString3d) {
+              let edges = value.points;
+              const endPoint = value.pointAt(0);
+              if (endPoint) {
+                edges = edges.concat([endPoint]);
+              }
+              builder.addLineString(edges);
+            } else if (value instanceof Arc3d) {
+              builder.addArc(value, false, false);
+            }
+          });
+        }
       } else if (geometry instanceof Path) {
         builder.addPath(geometry);
-      } else if (geometry instanceof Polyface) {
+      } else if (geometry instanceof IndexedPolyface) {
         builder.addPolyface(geometry, false);
+        if (styledGeometry.edges) {
+          // Since decorators don't natively support visual edges,
+          // We draw them manually as lines along each facet edge
+          // Need to offset polyface edges to ensure they don't overlap with the shape itself
```

Comment:
I think if you supplied the `pickableId` argument to `createGraphicBuilder`, you would not need to mess around with offsetting your edges.